### PR TITLE
Implement keyboard navigation for the dxDropDownBox widget

### PR DESCRIPTION
**Pull request by Yarovaya**

### CODE REVIEW

File: js/ui/drop_down_box.js [hunk 1]
```diff
@@ -20,6 +22,41 @@ var DROP_DOWN_BOX_CLASS = "dx-dropdownbox",
...
+            var $element = getLast ? $elements[length - i - 1] : $elements[i];
+
+            if(!tabbable && selectors.tabbable(null, $element)) {
+                tabbable = $element;
+                break;
+            }
+        }
+
+        return tabbable;
+    },
```
Comment (vconst) [inline js/ui/drop_down_box.js]: I think that this method may be simpler:

_getTabbable: function(getLast) {
    var $tabbableElements = this.content().find(selectors.tabbable);
    return getLast ? $tabbableElements.last() : $tabbableElements.first();
}

Comment (vconst) [reply js/ui/drop_down_box.js]: Also, I think that _getTabbable method should be replaced with the _getTabbableElements method. And first/last methods may be called in several places where it is needed

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/drop_down_box.js [hunk 2]
```diff
@@ -20,6 +22,29 @@ var DROP_DOWN_BOX_CLASS = "dx-dropdownbox",
...
+                    $focusableElement = e.shiftKey ? $tabbableElements.last() : $tabbableElements.first();
+
+                $focusableElement && $focusableElement.focus();
+                e.preventDefault();
+            }
+        });
+    },
+
+    _getTabbableElements: function() {
+        return this._getElements().filter(selectors.tabbable);
```
Comment (vconst) [inline js/ui/drop_down_box.js:42]: Change this._getElements().filter(selectors.tabbable) to this.content().find(selectors.tabbable) and remove unnecessary _getElements method.

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/drop_down_box.js [hunk 3]
```diff
@@ -20,6 +22,44 @@ var DROP_DOWN_BOX_CLASS = "dx-dropdownbox",
...
+
+            if(tabbable) {
+                break;
+            }
+        }
+
+        return tabbable;
+    },
+
+    _getElements: function() {
```
Comment (lukyanovas) [inline js/ui/drop_down_box.js]: I think this method is useless

**Review by vconst — APPROVED** [review]

**Pull request merged**